src/__tests__/index.spec.ts

```diff
@@ -210,3 +207,236 @@ describe('JsonMix - collections', function() {
+    const department: Department = new Department();
+    const value = new Department();
+    const result = await new JsonMix(data, department).withObject({ factory: () => value }).build();
+    expect(result).toBe(department);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+  });
+
+  it('Mix single data object with prototype (options.factory => promise)', async function() {
+    const department: Department = new Department();
+    const promise = Promise.resolve(new Department());
+    const result = await new JsonMix(data, department).withObject({ factory: () => promise }).build();
+    expect(result).toBe(department);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+  });
+
+  it('Mix single data object with prototype (options.factory => observable)', async function() {
+    const department: Department = new Department();
+    const observable = of(new Department());
+    const result = await new JsonMix(data, department).withObject({ factory: () => observable }).build();
+    expect(result).toBe(department);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+  });
+
+  it('Mix single data object with prototype inside another object', async function() {
+    const department: Department = new Department();
+    await new JsonMix(data, department).withObject(Pet, 'pet').build();
+    expect((department as any).pet.getLike()).toContain('Tom likes milk');
+  });
+
+  it('Mix multiple objects', async function() {
+    const department: Department = new Department();
+    const result = await new JsonMix(data, department)
+      .withObject(Department)
+      .withObject(Pet, 'pet')
+      .build();
+    expect(result).toBe(department);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+    expect((department as any).pet.getLike()).toContain('Tom likes milk');
+  });
+
+  it('Mix multiple objects in inverse order', async function() {
+    const department: Department = new Department();
+    const result = await new JsonMix(data, department)
+      .withObject(Pet, 'pet')
+      .withObject(Department)
+      .build();
+    expect(result).toBe(department);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+    expect((department as any).pet.getLike()).toContain('Tom likes milk');
+  });
+
+  it('Mix multiple objects ', async function() {
+    const department: Department = new Department();
+    const pet: Pet = new Pet();
+    (department as any).pet = pet;
+    const result = await new JsonMix(data, department)
+      .withObject(Pet, 'pet')
+      .withObject(Department)
+      .build();
+    expect(result).toBe(department);
+    expect((department as any).pet).toBe(pet);
+    expect(department.getTime()).toContain('Time at Center of the World is');
+    expect((department as any).pet.getLike()).toContain('Tom likes milk');
+  });
+});
```

Comment:
Need  a few tests where the object we pass in already has data. When it's not overridden, when it's overridden, when it's merged, with arrays/objects, etc.